src/Plugin/Queue/Handlers/Source/DropSourceHandler.php

```diff
@@ -0,0 +1,152 @@
+<?php declare(strict_types=1);
+
+/*
+  Copyright (c) 2023, Manticore Software LTD (https://manticoresearch.com)
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License version 2 or any later
+  version. You should have received a copy of the GPL license along with this
+  program; if you did not, you can find it at http://www.gnu.org/
+*/
+
+namespace Manticoresearch\Buddy\Base\Plugin\Queue\Handlers\Source;
+
+use Manticoresearch\Buddy\Base\Plugin\Queue\Handlers\BaseDropHandler;
+use Manticoresearch\Buddy\Base\Plugin\Queue\Payload;
+use Manticoresearch\Buddy\Core\Error\ManticoreSearchClientError;
+use Manticoresearch\Buddy\Core\ManticoreSearch\Client;
+
+/**
+ * @extends BaseDropHandler<array{
+ *       DROP: array{
+ *           expr_type?: string,
+ *           option: bool,
+ *           if-exists: bool,
+ *           sub_tree: array{
+ *               array{
+ *                   expr_type: string,
+ *                   base_expr: string
+ *               },
+ *               array{
+ *                   expr_type: string,
+ *                   base_expr: string,
+ *                   sub_tree: array{
+ *                      array{
+ *                        expr_type: string,
+ *                        table: string,
+ *                        no_quotes: array{
+ *                            delim: bool,
+ *                            parts: array<string>
+ *                        },
+ *                        alias: bool,
+ *                        base_expr: string,
+ *                        delim: bool
+ *                      }
+ *                   }
+ *               }
+ *           }
+ *       }
+ *   }>
+ */
+final class DropSourceHandler extends BaseDropHandler {
+
+	/**
+	 * @param string $name
+	 * @param string $tableName
+	 * @return int
+	 * @throws ManticoreSearchClientError
+	 */
+	protected function processDrop(string $name, string $tableName): int {
+		$manticoreClient = $this->manticoreClient;
+		$sql = /** @lang Manticore */
+			"SELECT * FROM $tableName WHERE match('@name \"$name\"')";
+
+
+		$result = $manticoreClient->sendRequest($sql);
+
+		if ($result->hasError()) {
+			throw ManticoreSearchClientError::create((string)$result->getError());
+		}
+
+		$removed = 0;
+		$result = $result->getResult();
+		if (is_array($result[0])) {
+			foreach ($result[0]['data'] as $sourceRow) {
+				$this->payload::$processor
+					->execute('stopWorkerById', [$sourceRow['full_name']]);
+
+				self::removeSourceRowData($sourceRow, $manticoreClient);
+				$removed++;
+			}
+		}
+
```

Comment:
It's a good idea to avoid multiple empty lines

Reply:
I'm not sure about it. multiple empty lines can be some logical delimiter in code, allow it easier to understand. However in our code-style it's not mandatory